### PR TITLE
Fix murcko old main

### PR DESCRIPTION
**Pull request by MSchokolowski**

### CODE REVIEW

File: src/lib/coaler/core/Matcher.cpp [hunk 1]
```diff
@@ -5,15 +5,71 @@
...
 
+#include <queue>
+
 #include "GraphMol/ChemTransforms/ChemTransforms.h"
 #include "GraphMol/DistGeomHelpers/Embedder.h"
 #include "GraphMol/FMCS/FMCS.h"
 
 namespace coaler::core {
+
+    void Matcher::murckoPruningRecursive(RDKit::RWMOL_SPTR mol, int atomID, int parentID, std::vector<bool>& visit,
```
Comment (cnoahris) [inline src/lib/coaler/core/Matcher.cpp:23]: reference the shared_ptr instead of copying it

File: src/lib/coaler/core/Matcher.cpp [hunk 2]
```diff
@@ -5,15 +5,71 @@
...
+                                         std::vector<int>& delAtoms, std::vector<std::pair<int, int>>& delBonds,
+                                         std::vector<int>& ringAtoms) {
+        // mark atom as visited and find all neighbor atoms of atom with atomID
+        visit.at(atomID) = true;
+        for (const auto& nbri : boost::make_iterator_range(mol->getAtomNeighbors(mol->getAtomWithIdx(atomID)))) {
```
Comment (cnoahris) [inline src/lib/coaler/core/Matcher.cpp:28]: meaning of nbri is unclear.

File: test/test_core.cpp [hunk 3]
```diff
@@ -24,4 +26,22 @@ TEST_CASE("Core_constructor", "[core]") {
     //
     //     auto coreScaffoldMurcko = Matcher::calculateCoreMurcko(mols);
     //     CHECK(RDKit::MolToSmarts(*coreScaffoldMurcko.value().first) == "[#6]1:[#6]:[#6]:[#6]:[#6]:[#6]:1");
+}
+
+TEST_CASE("Core_Murcko", "[core]") {
```
Comment (cnoahris) [inline test/test_core.cpp:31]: preferably use sections

File: src/lib/coaler/core/Matcher.cpp [hunk 4]
```diff
@@ -5,15 +5,71 @@
...
+            // atoms with only one neighbor (parent) will be deleted after recursive call.
+            // They are not deleted here cause internal IDs are updated after deletion.
+            if (numNbrs == 1) {
+                delBonds.push_back(std::make_pair(nbri, atomID));
+                delAtoms.push_back(nbri);
+            }
+        }
+    }
+
+    void Matcher::murckoCheckDelAtoms(RDKit::RWMOL_SPTR mol, int atomID, int parentID, std::vector<bool>& visit,
```
Comment (cnoahris) [inline src/lib/coaler/core/Matcher.cpp:54]: const ref to shared_ptr

**Review by cnoahris — COMMENTED** [review]
very nice results. only few make something const ref requests :)

File: src/lib/coaler/core/Matcher.cpp (hunk 2 above)
Comment (MSchokolowski) [reply src/lib/coaler/core/Matcher.cpp:28]: ist neighborID, ändere ich nochmal

**Review by cnoahris — APPROVED** [review]

**Pull request merged**